src/test/java/nl/tudelft/scrumbledore/LoggerTest.java

```diff
@@ -68,8 +70,8 @@ public final void testLog() {
     logger.log("log_write_test");
 
     try {
-      BufferedReader reader = new BufferedReader(
-          new InputStreamReader(new FileInputStream(loggingFile), "UTF-8"));
+      BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(
```

Comment:
This test case should not be reading files by itself anymore. Now that you have the getFirstLine and getLastLine methods, you do not need to do any I/O in the test suite, which is bad practice since it requires these try-catch statements. Can you refactor it to not have any try-catch, nor while loops et cetera?
<testing